packages/mobile/src/paymentRequest/PaymentRequestConfirmation.tsx

```diff
@@ -23,86 +22,69 @@ import { StackParamList } from 'src/navigator/types'
 import { writePaymentRequest } from 'src/paymentRequest/actions'
 import { PaymentRequestStatus } from 'src/paymentRequest/types'
 import { getDisplayName } from 'src/recipients/recipient'
-import { RootState } from 'src/redux/reducers'
-import { ConfirmationInput, getConfirmationInput } from 'src/send/utils'
+import useSelector from 'src/redux/useSelector'
+import { getConfirmationInput } from 'src/send/utils'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import Logger from 'src/utils/Logger'
-import { currentAccountSelector } from 'src/web3/selectors'
+import { walletAddressSelector } from 'src/web3/selectors'
 
 // @ts-ignore
 const TAG = 'paymentRequest/confirmation'
 
-interface StateProps {
-  e164PhoneNumber: string | null
-  account: string | null
-  confirmationInput: ConfirmationInput
-  addressJustValidated?: boolean
-}
+type OwnProps = StackScreenProps<StackParamList, Screens.PaymentRequestConfirmation>
 
-interface DispatchProps {
-  showError: typeof showError
-  writePaymentRequest: typeof writePaymentRequest
-}
+type Props = WithTranslation & OwnProps
+
+export const paymentConfirmationScreenNavOptions = () => ({
+  ...emptyHeader,
+  headerLeft: () => <BackButton eventName={RequestEvents.request_confirm_back} />,
+})
 
-const mapDispatchToProps = { showError, writePaymentRequest }
+function PaymentRequestConfirmation({ route }: Props) {
+  const [comment, setComment] = useState('')
+  const [transactionData] = useState(route.params.transactionData)
```

Comment:
For the values that come from `routes` and don't change the `useState` hook isn't really adding much value. We can simply do:
```
const { transactionData, addressJustValidated } = route.params
```
which does the same thing without the hook :)